sources/packages/web/src/views/aest/student/CASSupplierInformation.vue

```diff
@@ -0,0 +1,147 @@
+            <template #[`item.dateCreated`]="{ item }">
+              {{ dateOnlyLongString(item.dateCreated) }}
+            </template>
+            <template #[`item.supplierNumber`]="{ item }">
+              {{ item.supplierNumber }}
+            </template>
+            <template #[`item.supplierProtected`]="{ item }">
+              {{ booleanToYesNo(item.supplierProtected) }}
+            </template>
+            <template #[`item.supplierStatus`]="{ item }">
+              <status-chip-supplier :status="item.supplierStatus" />
+            </template>
+            <template #[`item.isValid`]="{ item }">
+              {{ booleanToYesNo(item.isValid) }}
+            </template>
+            <template #[`item.supplierSiteCode`]="{ item }">
+              {{ item.supplierSiteCode }}
+            </template>
+            <template #[`item.addressLine1`]="{ item }">
+              {{ emptyStringFiller(item.addressLine1) }}
+            </template>
+            <template #[`item.siteStatus`]="{ item }">
+              {{ emptyStringFiller(item.siteStatus) }}
+            </template>
+            <template #[`item.siteProtected`]="{ item }">
+              {{ emptyStringFiller(item.siteProtected) }}
+            </template>
+          </v-data-table>
+        </toggle-content>
+      </content-group>
+      <edit-c-a-s-supplier-info-modal
+        ref="addCASSupplierModal"
+        :allowedRole="Role.AESTEditCASSupplierInfo"
+      />
+    </body-header-container>
+  </tab-container>
+</template>
+
+<script lang="ts">
+import { ref, defineComponent, watchEffect } from "vue";
+import {
+  DEFAULT_PAGE_LIMIT,
+  Role,
+  CASSupplierInformationHeaders,
+} from "@/types";
+import { ModalDialog, useSnackBar, useFormatters } from "@/composables";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+import EditCASSupplierInfoModal from "@/components/aest/students/EditCASSupplierInfoModal.vue";
+import {
+  AddCASSupplierAPIInDTO,
+  CASSupplierInfoAPIOutDTO,
+} from "@/services/http/dto";
+import { CASSupplierService } from "@/services/CASSupplierService";
+import StatusChipSupplier from "@/components/generic/StatusChipSupplier.vue";
+
+export default defineComponent({
+  components: {
+    CheckPermissionRole,
+    EditCASSupplierInfoModal,
+    StatusChipSupplier,
+  },
+  props: {
+    studentId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props) {
+    const { dateOnlyLongString, emptyStringFiller } = useFormatters();
+    const { booleanToYesNo } = useFormatters();
+    const showModal = ref(false);
+    const casSupplierInfo = ref({} as CASSupplierInfoAPIOutDTO);
+    const addCASSupplierModal = ref(
+      {} as ModalDialog<AddCASSupplierAPIInDTO | boolean>,
+    );
+    const snackBar = useSnackBar();
+    const loadCASSuppliers = async (studentId: number) => {
+      casSupplierInfo.value =
+        await CASSupplierService.shared.getSupplierInfoByStudentId(studentId);
+    };
```

Comment:
Because we do not have a glocal API error handling, can we have a try-catch to display and unexpected error toast?